File: 9. Week 9 - Maps/solutions/python/visualization.py
import json
import matplotlib.pyplot as plt
import os

def visualize():
    data_path = '../../data/runs.geojson'
    
    if not os.path.exists(data_path):
        print(f"Data not found at {data_path}")
        return

    with open(data_path, 'r') as f:
        data = json.load(f)
    
    plt.figure(figsize=(10, 10))
    
    # Iterate through features and plot LineStrings
    for feature in data['features']:
        geom = feature['geometry']
        if geom['type'] == 'LineString':
            coords = geom['coordinates']
            # GeoJSON is [lon, lat], matplotlib needs x (lon), y (lat) lists
            lons = [c[0] for c in coords]
            lats = [c[1] for c in coords]
            
            plt.plot(lons, lats, color='#e74c3c', alpha=0.5, linewidth=1)
            
    plt.title('Run Routes Map', fontsize=16)
    plt.axis('equal') # Standard for maps
    plt.axis('off')   # Clean look
    
    plt.tight_layout()
    plt.savefig('map.png')
    print("Map saved to map.png")
    plt.show()

if __name__ == "__main__":
    visualize()

File: 9. Week 9 - Maps/solutions/r/visualization.R
# Visualize GeoJSON Maps using ggplot2 and sf

if (!require(ggplot2)) install.packages("ggplot2")
if (!require(sf)) install.packages("sf") # For GeoJSON handling
library(ggplot2)
library(sf)

data_path <- "../../data/runs.geojson"

if (file.exists(data_path)) {
  # Load GeoJSON
  runs_sf <- st_read(data_path, quiet = TRUE)
  
  # Plot
  # geom_sf handles projection and coordinate plotting automatically
  p <- ggplot(runs_sf) +
    geom_sf(color = "#e74c3c", linewidth = 0.5, alpha = 0.6) +
    
    # Theme
    theme_void() + # Maps look best clean
    labs(title = "Run Routes Map") +
    theme(plot.title = element_text(hjust = 0.5, size = 16))
  
  ggsave("map.png", plot = p, width = 10, height = 10)
  print("Map saved to map.png")

} else {
  print(paste("Error: File not found at", data_path))
}
